opal_server/pubsub.py

```diff
@@ -35,19 +41,25 @@ def __init__(self, signer: JWTSigner, broadcaster_uri: str = None):
         authenticator = WebsocketJWTAuthenticator(signer)
 
         @self.router.websocket("/ws")
-        async def websocket_rpc_endpoint(websocket: WebSocket, logged_in: bool = Depends(authenticator)):
+        async def websocket_rpc_endpoint(websocket: WebSocket, claims: JWTClaims = Depends(authenticator)):
             """
             this is the main websocket endpoint the sidecar uses to register on policy updates.
             as you can see, this endpoint is protected by an HTTP Authorization Bearer token.
             """
-            if not logged_in:
+            if claims is None:
                 logger.info("Closing connection, remote address: {remote_address}",
                             remote_address=websocket.client, reason="Authentication failed")
                 await websocket.close()
                 return
             # Init PubSub main-loop with or without broadcasting
             if broadcaster_uri is not None:
                 async with self.endpoint.broadcaster:
-                    await self.endpoint.main_loop(websocket)
+                    await self.endpoint.main_loop(websocket, claims=claims)
             else:
-                await self.endpoint.main_loop(websocket)
+                await self.endpoint.main_loop(websocket, claims=claims)
+
+    def _verify_permitted_topics(self, topics: Union[TopicList, ALL_TOPICS], channel: RpcChannel):
+        if "permitted_topics" not in channel.context.get("claims", {}):
+            return
+        if set(topics).difference(channel.context["claims"]["permitted_topics"]):
+            raise Unauthorized(description=f"Invalid 'topics' to subscribe")
```

Comment:
Let's replace this with the other option you mentioned - so that the error to the client is more clear:
```
Client tried to subscribe to restricted topic: `bla`
```